scipy/stats/_morestats.py

```diff
@@ -2884,6 +2884,75 @@ def fligner(*samples, center='median', proportiontocut=0.05):
     pval = distributions.chi2.sf(Xsq, k - 1)  # 1 - cdf
     return FlignerResult(Xsq, pval)
 
+@_axis_nan_policy_factory(lambda x1: (x1,), n_samples=2, n_outputs=1)
```

Comment:
`apply_along_axis` doesn't really support paired input samples, so this was the quickest / easiest way. I can just iterate instead for simplicity.

Reply:
Oops. That's right. I forgot. I suppose the decorator provides a bit of value then. Just make sure it's not _too_ much slower than looping and this will be fine.

Reply:
It's not really ideal, but I think it isn't too bad: 

```
>>> x1.shape
(2, 5, 60)

# with @_axis_nan_policy_factory
>>> %timeit _mood_inner_lc(x1, x2, axis=2)
6.91 ms ± 38.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

# without decorator
>>> %%timeit 
>>> x = []
>>> for idx in set([(i[0], i[1]) for i in np.ndindex(x1.shape)]):
>>>     x.append(_mood_inner_lc(x1[idx], x2[idx]))
>>> res = np.reshape(x, x1.shape[:2])
6.46 ms ± 19.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
so 7% time increase.

Reply:
Cool, so if you think it makes the code cleaner, go ahead and use it. Up to you.